src/domains/dashboard/components/modal-portal.tsx

```diff
@@ -0,0 +1,20 @@
+import { ReactNode, useEffect, useRef } from "react"
+import { createPortal } from "react-dom"
+
+const modalRoot = document.getElementById("modal-root") as HTMLElement
```

Comment:
@Azein @builat perhaps we could use a similar approach for our modal dialogs as @jacekkolasa does here